wavefront/wavefront.go

```diff
@@ -56,16 +60,61 @@ type (
 
 // ensure that WavefrontAdapter implements the HandleMetricServiceServer interface.
 var _ metric.HandleMetricServiceServer = &WavefrontAdapter{}
+var hostTags map[string]string
 
 // createWavefrontReporter creates a reporter that periodically flushes metrics to Wavefront.
 func createWavefrontReporter(cfg *config.Params) {
-	hostTags := map[string]string{"source": cfg.Source}
+	hostTags = map[string]string{"source": cfg.Source}
 	if direct := cfg.GetDirect(); direct != nil {
 		go wf.WavefrontDirect(metrics.DefaultRegistry, cfg.FlushInterval, hostTags, cfg.Prefix, direct.Server, direct.Token)
 	} else if proxy := cfg.GetProxy(); proxy != nil {
 		addr, _ := net.ResolveTCPAddr("tcp", proxy.Address)
 		go wf.WavefrontProxy(metrics.DefaultRegistry, cfg.FlushInterval, hostTags, cfg.Prefix, addr)
 	}
+
+	fmt.Print("Preparing adapter metrics")
+	ticker := time.NewTicker(5 * time.Second)
```

Comment:
Declare a const for `5 * time.Second` at the top of the file and initialize the adapter metric reporter in a separate function.